src/content/platform-integration/ios/app-extensions.md

```diff
@@ -117,207 +159,341 @@ Choose one of the following sources for your data.
+
+    *   In the **Shell** text field, enter:
+
+        ```console
+        /bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" prepare
+        ```
+
+    *   Click **Close**.
+
+1.  (Xcode) Share the build configurations.
+
+    *   Open the **project navigator**.
+
+    *   In the main window under **PROJECT**, select
+        **Runner**.
+
+    *   Open the **Info** tab.
+
+    *   Expand **Configuration**.
+
+    *   Expand **Debug** and update the value for
+        **ShareExtension** to match the value for
+        **Runner**.
+
+    *   Repeat the previous step for **Profile**, and
+        **Release**.
 
-## Advanced scrolling behavior
+1.  (Xcode, Optional) Replace any storyboard files with
+    an extension class, if needed.
 
-By default, flutter view does not handle scroll gestures in a share extension. 
-To support a scrollable list in the share extension, follow [this instruction][]. 
+    *   Open the **project navigator**.
 
-[this instruction]: {{site.github}}/flutter/website/issues/164670#issuecomment-2740702986
+    *   Select **Runner** > **ShareExtension** > **Info**.
 
-## Test extensions
+    *   Expand **Information Property List**.
 
-Testing extensions on simulators and physical devices
-have slightly different procedures.
+    *   Delete the **NSExtensionMainStoryboard** key.
 
-{% comment %}
-The different procedures are necessary due to bugs(which bugs?) in Xcode.
-Revisit these docs after future Xcode releases to see if they are fixed.
-{% endcomment -%}
+    *   Add the **NSExtensionPrincipalClass** key.
 
-### Test on a simulator
+    *   Add one of these values for the
+        `NSExtensionPrincipalClass` key:
 
-1. Build and run the main application target.
-1. After the app is launched on the simulator,
-   press <kbd>Cmd</kbd> + <kbd>Shift</kbd> + <kbd>H</kbd>
-   to minimize the app, which switches to the home screen.
-1. Launch an app that supports the share extension,
-   such as the Photos app.
-1. Select a photo, tap the share button, then tap
-   on the share extension icon of your app.
+        *   (Swift) **ShareExtension.ShareViewController**
+        *   (Objective-C)  **ShareViewController**
 
-### Test on a physical device
+1.  (Xcode) Update the `ShareViewController` to use the
+    `FlutterViewController`.
 
-You can use the following procedure or the
-[Testing on simulators](#test-on-a-simulator) instructions
-to test on physical devices.
+    *   Open the **project navigator**.
+
+    *   Select **Runner** > **ShareExtension** > **ShareViewController**.
+
+    *   Update the `ShareViewController` file to use the
+        `FlutterViewController` class:
+
+{% tabs "register-plugins-tabs", true %}
+
+{% tab "UIKit-Swift" %}
+
+```swift title="ShareViewController.swift"
+import UIKit
+import Flutter
+
+class ShareViewController: UIViewController {
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        showFlutter()
```

Comment:
nit: maybe simplify it by not having this helper function.

Reply:
Want me to get rid of the `showFlutter()` function? Just to double-check... if I do this, will the sample app still open through the Share extension?

Reply:
oh i was talking about just: 

```
override func viewDidLoad() {
        super.viewDidLoad()
        let flutterViewController = FlutterViewController(project: nil, nibName: nil, bundle: nil)
        addChild(flutterViewController)
        view.addSubview(flutterViewController.view)
        flutterViewController.view.frame = view.bounds
}
```

Reply:
Ah, got it! My thoughts... I think we should keep the helper function, but just because it breaks up the code into smaller chunks. My hope is that those smaller chunks will be easier for a developer to scan over and interpret. Tentatively closing this one, but I get what you're saying.